opentelemetry-exporter-cloud-trace/src/opentelemetry/tools/resource_detector.py

```diff
@@ -8,27 +10,71 @@
 _GCP_METADATA_URL_HEADER = {"Metadata-Flavor": "Google"}
 
 
-def get_gce_resources():
-    """ Resource finder for common GCE attributes
-
-        See: https://cloud.google.com/compute/docs/storing-retrieving-metadata
-    """
+def _get_google_metadata_and_common_attributes():
     token = attach(set_value("suppress_instrumentation", True))
     all_metadata = requests.get(
         _GCP_METADATA_URL, headers=_GCP_METADATA_URL_HEADER
     ).json()
     detach(token)
-    gce_resources = {
-        "host.id": all_metadata["instance"]["id"],
+    common_attributes = {
         "cloud.account.id": all_metadata["project"]["projectId"],
-        "cloud.zone": all_metadata["instance"]["zone"].split("/")[-1],
         "cloud.provider": "gcp",
-        "gcp.resource_type": "gce_instance",
+        "cloud.zone": all_metadata["instance"]["zone"].split("/")[-1],
     }
-    return gce_resources
+    return common_attributes, all_metadata
+
+
+def get_gce_resources():
+    """ Resource finder for common GCE attributes
+
+        See: https://cloud.google.com/compute/docs/storing-retrieving-metadata
+    """
+    (
+        common_attributes,
+        all_metadata,
+    ) = _get_google_metadata_and_common_attributes()
+    common_attributes.update(
+        {
+            "host.id": all_metadata["instance"]["id"],
+            "gcp.resource_type": "gce_instance",
+        }
+    )
+    return common_attributes
+
+
+def get_gke_resources():
+    """ Resource finder for GKE attributes
+
+    """
+    # The user must specify these environment variables via the Downward API
+    container_name = os.getenv("CONTAINER_NAME")
+    pod_namespace = os.getenv("NAMESPACE")
+    if not container_name or not pod_namespace:
+        return {}
+    (
+        common_attributes,
+        all_metadata,
+    ) = _get_google_metadata_and_common_attributes()
+    common_attributes.update(
+        {
+            "k8s.cluster.name": all_metadata["instance"]["attributes"][
+                "cluster-name"
+            ],
+            "k8s.namespace.name": pod_namespace,
+            "host.id": all_metadata["instance"]["id"],
+            "k8s.pod.name": os.getenv("HOSTNAME", ""),
```

Comment:
It's being set by Kubernetes I believe, but it doesn't always 100% match the pod name due to trimming: https://github.com/kubernetes/kubernetes/issues/4825#issuecomment-186058020

So I think it would be safer to read a separate environment variable by default, e.g. `POD_NAME`, and possibly fallback to `HOSTNAME` as a best guess.

Reply:
Though this behavior does match the OpenCensus implementation: https://github.com/census-ecosystem/opencensus-go-resource/blob/87ca538ec76b3961839940406845ef624d794fa8/gke/gke.go#L48

/cc @rghetia see the issue above about trimming the pod name to 63 characters. Shall we introduce a separate environment variable to be passed via https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/#use-pod-fields-as-values-for-environment-variables ?